src/bpf/bpfd-manager.c

```diff
@@ -0,0 +1,448 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <sys/wait.h>
+
+#include "bpf-dlopen.h"
+#include "fileio.h"
+#include "sd-daemon.h"
+#include "sd-messages.h"
+#include "bpfd-manager.h"
+#include "fd-util.h"
+#include "fs-util.h"
+#include "hashmap.h"
+#include "missing_syscall.h"
+#include "mkdir.h"
+#include "process-util.h"
+#include "signal-util.h"
+#include "socket-util.h"
+#include "stdio-util.h"
+#include "umask-util.h"
+
+static int get_memlock_by_fd(int fd, char **memlock) {
+        _cleanup_free_ char *fdinfo = NULL;
+        char *memlock_line;
+        char path[STRLEN("/proc/self/fdinfo/") + DECIMAL_STR_MAX(int)];
+        int r;
+
+        xsprintf(path, "/proc/self/fdinfo/%i", fd);
+        r = read_full_virtual_file(path, &fdinfo, NULL);
+        if (r < 0)
+                return r;
+
+        memlock_line = find_line_startswith(fdinfo, "memlock:");
+        if (!memlock_line)
+                return -EOPNOTSUPP;
+
+        memlock_line += strspn(memlock_line, WHITESPACE);
+        memlock_line[strcspn(memlock_line, WHITESPACE)] = 0;
+
+        *memlock = strdup(memlock_line);
```

Comment:
coding style:

we have this rule that return params should not be touched on failure, and always be initialized on success. here you assign a value to the return parameter even on failure.

hence please, store result of strdup() in temporary var first, and only if it succeeded copy to the return parameter